src/AqueductV1Pair.sol

```diff
@@ -105,21 +159,117 @@ contract AqueductV1Pair is IAqueductV1Pair, AqueductV1ERC20, SuperAppBase {
             _reserve0 = uint256(int(Math.sqrt((_kLast * totalAmount0) / totalAmount1)) * (ePower + c) / (ePower - c));
             _reserve1 = _kLast / _reserve0;*/
         } else if (totalFlow0 > 0 && timeElapsed > 0) {
-            // use x * y = k
-            uint112 totalAmount0 = (totalFlow0 * timeElapsed * (10000 - UPPER_BOUND_FEE)) / 10000;
-            _reserve0 = reserve0 + totalAmount0;
-            _reserve1 = uint112(_kLast / _reserve0); // should be a safe downcast
+            (_reserve0, _reserve1) = _calculateReservesFlow0(_kLast, totalFlow0, timeElapsed, reserve0);
         } else if (totalFlow1 > 0 && timeElapsed > 0) {
-            // use x * y = k
-            uint112 totalAmount1 = (totalFlow1 * timeElapsed * (10000 - UPPER_BOUND_FEE)) / 10000;
-            _reserve1 = reserve1 + totalAmount1;
-            _reserve0 = uint112(_kLast / _reserve1); // should be a safe downcast
+            (_reserve0, _reserve1) = _calculateReservesFlow1(_kLast, totalFlow1, timeElapsed, reserve1);
         } else {
             // get static reserves
             (_reserve0, _reserve1, ) = getReserves();
         }
     }
 
+    /**
+     * @notice Calculates the reserve amount since the last update time after applying fees.
+     * @dev This function uses the total flow, elapsed time and the UPPER_BOUND_FEE to calculate the updated reserve amount.
+     * @param totalFlow The total flow of the token.
+     * @param timeElapsed The time elapsed since the last update.
+     * @return reserveAmountSinceTime calculated reserve amount after applying fees.
+     */
+    function _calculateReserveAmountSinceTime(
```

Comment:
This felt like a "suitable" name, but let me know if this doesn't make sense, was trying to guess some functionality along the way refactoring